codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -102,6 +105,7 @@ public void close() { }
     @Override
     public ChannelFuture writeData(ChannelHandlerContext ctx, int streamId, ByteBuf data,
             int padding, boolean endStream, ChannelPromise promise) {
+        SimpleChannelPromiseAggregator promiseAggregator = null;
```

Comment:
Right ... as I noted further down, we would need a second variable to keep track of when we're done writing everything for this aggregator.  Something like `aggregator.doneWriting()` (better name?)